### PR TITLE
Bump test-dapp to v7.1.0

### PR DESCRIPTION
**Pull request by seaona**
## Explanation
Bump test-dapp to v7.1.0
See changes [here](https://github.com/MetaMask/test-dapp/pull/254).

* Fixes https://github.com/MetaMask/MetaMask-planning/issues/1264

## Screenshots/Screencaps

https://github.com/MetaMask/metamask-extension/assets/54408225/0508d0e0-0b02-4f53-8a10-1f8ff3973856

### Before

`v7.0.1`

### After

`v7.1.0`

## Manual Testing Steps
- Check circle ci runs
- Locally run tests in your environment `yarn test:e2e:chrome`

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [X] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: test/e2e/nft/erc721-interaction.spec.js [hunk 1]
```diff
@@ -72,9 +72,12 @@ describe('ERC721 NFTs testdapp interaction', function () {
...
         // watch 3 of the nfts
-        await driver.clickElement({ text: 'Watch NFT 1', tag: 'button' });
-        await driver.clickElement({ text: 'Watch NFT 2', tag: 'button' });
-        await driver.clickElement({ text: 'Watch NFT 3', tag: 'button' });
+        await driver.fill('#watchNFTInput', '1');
+        await driver.clickElement({ text: 'Watch NFT', tag: 'button' });
+        await driver.fill('#watchNFTInput', '2');
+        await driver.clickElement({ text: 'Watch NFT', tag: 'button' });
+        await driver.fill('#watchNFTInput', '3');
+        await driver.clickElement({ text: 'Watch NFT', tag: 'button' });
```
Comment (seaona) [inline test/e2e/nft/erc721-interaction.spec.js:81]: we need to update this section as the test dapp layout has changed now, we don't have multiple buttons, but instead just 1 and and input field for setting up the token id

File: test/e2e/nft/erc721-interaction.spec.js [hunk 2]
```diff
@@ -98,9 +107,12 @@ describe('ERC721 NFTs testdapp interaction', function () {
...
         await driver.switchToWindowWithTitle('E2E Test Dapp', windowHandles);
-        await driver.clickElement({ text: 'Watch NFT 4', tag: 'button' });
-        await driver.clickElement({ text: 'Watch NFT 5', tag: 'button' });
-        await driver.clickElement({ text: 'Watch NFT 6', tag: 'button' });
+        await driver.fill('#watchNFTInput', '4');
+        await driver.clickElement({ text: 'Watch NFT', tag: 'button' });
+        await driver.fill('#watchNFTInput', '5');
+        await driver.clickElement({ text: 'Watch NFT', tag: 'button' });
+        await driver.fill('#watchNFTInput', '6');
+        await driver.clickElement({ text: 'Watch NFT', tag: 'button' });
```
Comment (seaona) [inline test/e2e/nft/erc721-interaction.spec.js:116]: same as above

**Review by DDDDDanica — APPROVED** [review]

**Comment by DDDDDanica** [comment]
LGTM !

**Review by legobeat — APPROVED** [review]
lgtm

File: test/e2e/nft/erc721-interaction.spec.js (hunk 1 above)
Comment (seaona) [reply test/e2e/nft/erc721-interaction.spec.js:81]: oh apologies I think the description was not 100% clear. So the update is already made (see right side), I just wanted explain why I was updating this section of test with the dapp bump :blush: 
Basically the test dapp has changed, and this part needed to be updated to make tests continue passing
does it make sense @legobeat ?

Comment (seaona) [reply test/e2e/nft/erc721-interaction.spec.js:81]: thank you v much for reviewing the comments too :heart_eyes:

**Pull request merged**